src/image_gen/imagegenutils.cpp

```diff
@@ -68,68 +68,130 @@ std::variant<absl::Status, std::optional<resolution_t>> getDimensions(const std:
-    auto it = payload.parsedJson->FindMember(keyName.c_str());
-    if (it == payload.parsedJson->MemberEnd()) {
+std::variant<absl::Status, std::optional<int64_t>> getInt64FromPayload(const ovms::MultiPartParser& payload, const std::string& keyName) {
+    std::string value = payload.getFieldByName(keyName);
+    if (value.empty()) {
+        return std::nullopt;
+    }
+    auto intValue = ovms::stoi64(value);
+    if (!intValue.has_value()) {
+        return absl::InvalidArgumentError(absl::StrCat(keyName, " field is not a int64"));
+    }
+    return intValue.value();
+}
+std::variant<absl::Status, std::optional<int>> getIntFromPayload(const rapidjson::Document& payload, const std::string& keyName) {
+    auto it = payload.FindMember(keyName.c_str());
+    if (it == payload.MemberEnd()) {
         return std::nullopt;
     }
     if (!it->value.IsInt()) {
         return absl::InvalidArgumentError(absl::StrCat(keyName, " field is not a int"));
     }
     return it->value.GetInt();
 }
-std::variant<absl::Status, std::optional<size_t>> getSizetFromPayload(const ovms::HttpPayload& payload, const std::string& keyName) {
-    auto it = payload.parsedJson->FindMember(keyName.c_str());
-    if (it == payload.parsedJson->MemberEnd()) {
+std::variant<absl::Status, std::optional<int>> getIntFromPayload(const ovms::MultiPartParser& payload, const std::string& keyName) {
+    std::string value = payload.getFieldByName(keyName);
+    if (value.empty()) {
+        return std::nullopt;
+    }
+    auto intValue = ovms::stoi32(value);
+    if (!intValue.has_value()) {
+        return absl::InvalidArgumentError(absl::StrCat(keyName, " field is not a int32"));
+    }
+    return intValue.value();
+}
+std::variant<absl::Status, std::optional<size_t>> getSizetFromPayload(const rapidjson::Document& payload, const std::string& keyName) {
+    auto it = payload.FindMember(keyName.c_str());
+    if (it == payload.MemberEnd()) {
         return std::nullopt;
     }
     if (!it->value.IsUint64()) {
         return absl::InvalidArgumentError(absl::StrCat(keyName, " field is not a size_t"));
     }
     return it->value.GetUint64();
 }
+std::variant<absl::Status, std::optional<size_t>> getSizetFromPayload(const ovms::MultiPartParser& payload, const std::string& keyName) {
+    std::string value = payload.getFieldByName(keyName);
+    if (value.empty()) {
+        return std::nullopt;
+    }
+    try {
+        size_t sizeValue = std::stoull(value);
+        return sizeValue;
+    } catch (const std::invalid_argument& e) {
+        return absl::InvalidArgumentError(absl::StrCat(keyName, " field is not a size_t: ", e.what()));
+    } catch (const std::out_of_range& e) {
+        return absl::InvalidArgumentError(absl::StrCat(keyName, " field is out of range for size_t: ", e.what()));
+    } catch (...) {
+        return absl::InvalidArgumentError(absl::StrCat(keyName, " field is not a size_t"));
+    }
```

Comment:
can also have a wrapper here as for previous cases? `ovms::stoull`

Reply:
good idea, done